src/workflows/context/context.py

```diff
@@ -535,6 +536,66 @@ async def my_step(self, ctx: Context, ev: StartEvent) -> WorkerEvent | GatherEve
 
         self._broker_log.append(message)
 
+    def send_events(self, events: list[Event], step: Optional[str] = None) -> None:
+        """
+        Emit events manually.
+
+        Args:
+            events (list[Event]): a list of Event objects to be sent
+            step (Optional[str]): a string representing the target step to which events should be sent. Defaults to None if not set.
+
+        Returns:
+            None
+
+        Examples:
+        ```
+        class MultipleEventsWorkflow(Workflow):
+        @step
+        async def send_events(self, ev: InputEvent, ctx: Context):
+            ctx.send_events(events = [OutputAEvent(), OutputBEvent()], step="step_a")
+        ```
+        """
+        for event in events:
+            self._events_queue[str(type(event))] += 1
+            if not step:
+                self._send_event(event)
+            else:
+                self._send_event(event, step)
+
+    def receive_events(
```

Comment:
I see in the decorator draft you're thinking about a `gather` parameter - let's call this function `gather_events` so the relation will be obvious.